CSS minification + index.html copying

resir014 opened the pull request:
* [x] Added CSS/SCSS minification on build w/ PostCSS.
* [x] Combine the `index.html` into the `dist` folder.
* [x] Minor styling changes in gulpfile.
* [x] ~~Update lockfile. (I don't have `node@^8.0.0` and `npm@^5.0.0` installed on my work laptop, so I can't update the lockfile. Will do that when I get home.)~~ Removed and gitignored as of d18bc20.

Some additional changes:

* Replaced `loader` with `use`, as it's deprecated for Webpack 3+

resir014 committed:
Added CSS/SCSS minification on build w/ PostCSS.

Some additional changes:

* Replaced `loader` with `use`, as it's deprecated for Webpack 3+

resir014 committed:
Removed todo item from readme.

resir014 commented:
On second thought, I'll put the second todo item up on this PR as well. Shouldn't be that hard.

South-Paw commented:
Minor styling change in the gulpfile added as a task too.

While airbnb style may not be a part of this repo, I'll probably update the gulpfile to follow it.

resir014 committed:
Minor styling fixes.

resir014 committed:
Fixed incorrect loader options causing errors in `babel-loader`

resir014 committed:
Combine index.html to dist/ with copy-webpack-plugin.

resir014 committed:
Clean `dist/` folder before proceeding with build task.

resir014 committed:
Fixed copy target path.

resir014 committed:
Use `cross-env` to easily set env variables across platforms

resir014 committed:
Removed todo item.

resir014 committed:
Deleted package-lock.json and gitignore it

Ideally, you should include this on your own repo (to make sure your team members have the same version of all packages), but we've decided not to include it here since it's a starter kit. Do comment this out when you're pushing to your own repo, though.

resir014 committed:
Removed `--config` flag on dev task

Webpack defaults to `webpack.config.js` so this is unnecessary.
